src/sdk/pynni/nni/nas/pytorch/utils.py

```diff
@@ -116,12 +116,14 @@ def update(self, val, n=1):
         n : int
             The weight of the new value.
         """
-        if not isinstance(val, float) and not isinstance(val, int):
+        if isinstance(val, float) or isinstance(val, int) \
+            or (isinstance(val, torch.Tensor) and not val.size()):
```

Comment:
`torch.is_tensor(val)`. Also I believe we don't have to limit usage of AverageMeter to number and single-value tensor. Arrays and tensors can also use `AverageMeter` if users want to. In fact, I don't think this `if` is necessary at all.